src/solver/expressions/visitors/PortfieldSubstitutionVisitor.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+** Copyright 2007-2024, RTE (https://www.rte-france.com)
+** See AUTHORS.txt
+** SPDX-License-Identifier: MPL-2.0
+** This file is part of Antares-Simulator,
+** Adequacy and Performance assessment for interconnected energy networks.
+**
+** Antares_Simulator is free software: you can redistribute it and/or modify
+** it under the terms of the Mozilla Public Licence 2.0 as published by
+** the Mozilla Foundation, either version 2 of the License, or
+** (at your option) any later version.
+**
+** Antares_Simulator is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** Mozilla Public Licence 2.0 for more details.
+**
+** You should have received a copy of the Mozilla Public Licence 2.0
+** along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+*/
+#include <antares/solver/expressions/nodes/ExpressionsNodes.h>
+#include <antares/solver/expressions/visitors/PortfieldSubstitutionVisitor.h>
+
+namespace Antares::Solver::Visitors
+{
+
+PortfieldSubstitutionVisitor::PortfieldSubstitutionVisitor(Registry<Nodes::Node>& registry,
+                                                           PortfieldSubstitutionContext& ctx):
+    CloneVisitor(registry),
+    ctx_(ctx)
+{
+}
+
+Nodes::Node* PortfieldSubstitutionVisitor::visit(const Nodes::PortFieldNode* node)
+{
+    if (ctx_.portfield.contains(*node))
+    {
+        return (ctx_.portfield.at(*node));
+    }
```

Comment:
### Performance

`std::unordered_map::contains` is nice, but here we perform 2 lookup operations
1. Check if `ctx_.portfield` contains `*node`
2. If so, lookup the the corresponding value and return it

I think we'd better stick to the good old `std::unordered_map::find`, to get the iterator, check it against `ctx_.portfield.end()`, etc.

According to the doc, [`contains`](https://en.cppreference.com/w/cpp/container/unordered_map/contains), [`at`](https://en.cppreference.com/w/cpp/container/unordered_map/at) and [`find`](https://en.cppreference.com/w/cpp/container/unordered_map/find) are all constant on average, worst case linear in the size of the container.